src/frameworks/compose/discover/filesystem.ts

```diff
@@ -0,0 +1,59 @@
+import { FileSystem } from "./types";
+import { pathExists, readFile } from "fs-extra";
+import { logger } from "../../..";
+import * as path from "path";
+
+/**
+ * Find files or read contents present in the Repository
```

Comment:
what does "Repository" mean?

I'm not sure why this class is just called "LocalFilesystem" 🤔

I think the intention for defining the `FileSystem` was to allow us to more easily test our discovery code (e.g. create `InMemoryFileSystem"). 

I like that idea, but if we aren't leveraging that kind of design, I think this interface-based design is pure overhead.

Reply:
Since we link Github repo, I kept it as "Repository". 
Do you mean why this class isn't called "LocalFilesystem"?

I will create an `InMemoryFileSystem` for this.